seerep_srv/seerep_core/include/seerep_core/core_dataset.h

```diff
@@ -290,22 +290,40 @@ class CoreDataset
                         std::optional<std::vector<seerep_core_msgs::AabbTimeIdPair>>& timetree_result,
                         std::optional<std::set<boost::uuids::uuid>>& semanticResult,
                         std::optional<std::vector<boost::uuids::uuid>>& instanceResult,
-                        const std::optional<std::vector<boost::uuids::uuid>>& dataUuids);
-
+                        const std::optional<std::vector<boost::uuids::uuid>>& dataUuids, bool sortByTime = false);
+  /**
+   * @brief sorts the uuids of the result set based on their timestamp
+   *
+   * @param timetree_result the result from the timetree query. Contains uuids and the corresponding timestamp
+   * @param intersectedResult the intersected uuids of all parts of the query
+   * @return std::vector<boost::uuids::uuid> uuids of the result set sorted by timestamp
+   */
+  std::vector<boost::uuids::uuid>
+  sortResultByTime(std::optional<std::vector<seerep_core_msgs::AabbTimeIdPair>>& timetree_result,
+                   std::set<boost::uuids::uuid> intersectionResult);
+  /**
+   * @brief does the actual sorting for the sortResultByTime method
+   *
+   * @param sortingVector the data to be sorted
+   * @return std::vector<boost::uuids::uuid> sorted vector of result uuids
+   */
+  std::vector<boost::uuids::uuid>
+  sortResultByTimeActualSorting(std::vector<std::pair<int64_t, boost::uuids::uuid>>& sortingVector);
```

Comment:
Combine these functions? I don't think it makes sense to call them individually, and the naming is not the best.